Source/Core/VideoBackends/OGL/ProgramShaderCache.cpp

```diff
@@ -414,79 +423,40 @@ void ProgramShaderCache::Init()
 
-  // Read our shader cache, only if supported
-  if (g_ogl_config.bSupportsGLSLCache)
-  {
-    GLint Supported;
-    glGetIntegerv(GL_NUM_PROGRAM_BINARY_FORMATS, &Supported);
-    if (!Supported)
-    {
-      ERROR_LOG(VIDEO, "GL_ARB_get_program_binary is supported, but no binary format is known. So "
-                       "disable shader cache.");
-      g_ogl_config.bSupportsGLSLCache = false;
-    }
-    else
-    {
-      if (!File::Exists(File::GetUserPath(D_SHADERCACHE_IDX)))
-        File::CreateDir(File::GetUserPath(D_SHADERCACHE_IDX));
-
-      std::string cache_filename =
-          StringFromFormat("%sogl-%s-shaders.cache", File::GetUserPath(D_SHADERCACHE_IDX).c_str(),
-                           SConfig::GetInstance().m_strUniqueID.c_str());
-
-      ProgramShaderCacheInserter inserter;
-      g_program_disk_cache.OpenAndRead(cache_filename, inserter);
-    }
-    SETSTAT(stats.numPixelShadersAlive, pshaders.size());
-  }
-
   CreateHeader();
 
   CurrentProgram = 0;
   last_entry = nullptr;
 }
 
+void ProgramShaderCache::LoadCache()
+{
+  // Read our shader cache
+  if (!File::Exists(File::GetUserPath(D_SHADERCACHE_IDX)))
+    File::CreateDir(File::GetUserPath(D_SHADERCACHE_IDX));
+
+  std::string cache_filename =
+      StringFromFormat("%sogl-%s-shaders.cache", File::GetUserPath(D_SHADERCACHE_IDX).c_str(),
+                       SConfig::GetInstance().m_strUniqueID.c_str());
+
+  ProgramShaderCacheInserter inserter;
+  g_program_disk_cache.OpenAndRead(cache_filename, inserter);
+}
+
 void ProgramShaderCache::Shutdown()
 {
   // store all shaders in cache on disk
-  if (g_ogl_config.bSupportsGLSLCache)
+  for (auto& entry : pshaders)
   {
-    for (auto& entry : pshaders)
+    if (!entry.second.in_cache)
     {
-      // Clear any prior error code
-      glGetError();
-
-      if (entry.second.in_cache)
-      {
-        continue;
-      }
-
-      GLint link_status = GL_FALSE, delete_status = GL_TRUE, binary_size = 0;
-      glGetProgramiv(entry.second.shader.glprogid, GL_LINK_STATUS, &link_status);
-      glGetProgramiv(entry.second.shader.glprogid, GL_DELETE_STATUS, &delete_status);
-      glGetProgramiv(entry.second.shader.glprogid, GL_PROGRAM_BINARY_LENGTH, &binary_size);
-      if (glGetError() != GL_NO_ERROR || link_status == GL_FALSE || delete_status == GL_TRUE ||
-          !binary_size)
-      {
-        continue;
-      }
-
-      std::vector<u8> data(binary_size + sizeof(GLenum));
-      u8* binary = &data[sizeof(GLenum)];
-      GLenum* prog_format = (GLenum*)&data[0];
-      glGetProgramBinary(entry.second.shader.glprogid, binary_size, nullptr, prog_format, binary);
-      if (glGetError() != GL_NO_ERROR)
-      {
-        continue;
-      }
-
-      g_program_disk_cache.Append(entry.first, &data[0], binary_size + sizeof(GLenum));
+      g_program_disk_cache.Append(entry.first, nullptr, 0);
     }
-
-    g_program_disk_cache.Sync();
-    g_program_disk_cache.Close();
   }
 
+  g_program_disk_cache.Sync();
```

Comment:
considering the LinearDiskCache is implemented with an fstream, just calling Close() will be equivalent to doing a Sync(), since an fstream flushes anything pending before closing.